src/astarte_storage.c

```diff
@@ -0,0 +1,315 @@
+/*
+ * (C) Copyright 2018-2023, SECO Mind Srl
+ *
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR Apache-2.0
+ */
+
+#include "astarte_storage.h"
+
+#include <esp_log.h>
+#include <nvs.h>
+#include <stdlib.h>
+#include <string.h>
+
+/************************************************
+ *        Defines, constants and typedef        *
+ ***********************************************/
+
+#define TAG "ASTARTE_STORAGE"
+
+#define NVS_NAMESPACE "ASTARTE_STORAGE"
+
+/************************************************
+ *         Global functions definitions         *
+ ***********************************************/
+
+astarte_storage_err_t astarte_storage_store_property(
+    const char *interface_name, const char *path, const void *data, size_t data_len)
+{
+    if (data == NULL) {
+        return astarte_storage_delete_property(interface_name, path);
+    }
+
+    nvs_handle_t handle = 0U;
+    esp_err_t esp_err = nvs_open(NVS_NAMESPACE, NVS_READWRITE, &handle);
+    if (esp_err != ESP_OK) {
+        return ASTARTE_STORAGE_ERR_INTERNAL;
+    }
+
+    // Get the full key interface_name + path
+    size_t interface_name_len = strlen(interface_name);
+    size_t path_len = strlen(path);
+    char *key = calloc(interface_name_len + path_len + 1, sizeof(char));
+    if (key == NULL) {
+        return ASTARTE_STORAGE_ERR_INTERNAL;
+    }
+    strcat(strcat(key, interface_name), path);
```

Comment:
We cannot be sure that the `interface_name` and `path` params were passed as a null-terminated string, so it is better to use `strncat` function.